src/HexagonOptimize.cpp

```diff
@@ -214,12 +220,17 @@ Expr apply_patterns(Expr x, const std::vector<Pattern> &patterns, IRMutator *op_
                 if (p.flags & Pattern::SwapOperands) {
                     std::swap(matches[0], matches[1]);
                 }
-                if (p.flags & Pattern::LosslessCast) {
+                if ((p.flags & Pattern::LosslessCastOp1) ||
+                    (p.flags & Pattern::LosslessCastOp0)) {
                     internal_assert(matches.size() == 2);
-                    Type t = Type(matches[1].type().code(), matches[1].type().bits()/2, 1);
-                    Expr b = lossless_cast(t, matches[1]);
+                    // LosslessCastOpx is a way of knowing that operand x is a broadcast.
+                    // Therefore, both *_should_not_* be set in the pattern flags because
+                    // that'll mean we are operating on two broadcasts.
+                    int op_num = (p.flags & Pattern::LosslessCastOp0) ? 0 : 1;
+                    Type t = Type(matches[op_num].type().code(), matches[op_num].type().bits()/2, 1);
```

Comment:
Type has with_* methods that make this easier (IMO):

    Type t = matches[op_num].type();
    t = t.with_bits(t.bits() / 2).with_lanes(1);